src/ini.cpp

```diff
@@ -91,30 +93,11 @@ bool IniFile::SaveToDisk(const std::string &filename)
 	if (ret != 0) return false;
 #endif
 
-#if defined(_WIN32)
-	/* Allocate space for one more \0 character. */
-	wchar_t tfilename[MAX_PATH + 1], tfile_new[MAX_PATH + 1];
-	wcsncpy(tfilename, OTTD2FS(filename).c_str(), MAX_PATH);
-	wcsncpy(tfile_new, OTTD2FS(file_new).c_str(), MAX_PATH);
-	/* SHFileOperation wants a double '\0' terminated string. */
-	tfilename[MAX_PATH - 1] = '\0';
-	tfile_new[MAX_PATH - 1] = '\0';
-	tfilename[wcslen(tfilename) + 1] = '\0';
-	tfile_new[wcslen(tfile_new) + 1] = '\0';
-
-	/* Rename file without any user confirmation. */
-	SHFILEOPSTRUCT shfopt;
-	MemSetT(&shfopt, 0);
-	shfopt.wFunc  = FO_MOVE;
-	shfopt.fFlags = FOF_NOCONFIRMATION | FOF_NOCONFIRMMKDIR | FOF_NOERRORUI | FOF_SILENT;
-	shfopt.pFrom  = tfile_new;
-	shfopt.pTo    = tfilename;
-	SHFileOperation(&shfopt);
-#else
-	if (rename(file_new.c_str(), filename.c_str()) < 0) {
-		Debug(misc, 0, "Renaming {} to {} failed; configuration not saved", file_new, filename);
+	std::error_code ec;
+	std::filesystem::rename(file_new, filename, ec);
```

Comment:
Since we use OTTD2FS for the `std::ofstream` (line 58), I guess we should also use it here.

Reply:
std::filesystem::path should be handle to handle conversion, but we dont't/can't generally use u8string/char8_t.

So I've put OTTD2FS in here, on the assumption it's needed.